app/src/main/java/org/oppia/app/player/exploration/ExplorationManagerFragmentPresenter.kt

```diff
@@ -18,18 +19,19 @@ import javax.inject.Inject
 class ExplorationManagerFragmentPresenter @Inject constructor(
   private val profileManagementController: ProfileManagementController,
   private val logger: Logger,
-  private val activity: AppCompatActivity
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment
 ) {
   private lateinit var profileId: ProfileId
 
   fun handleCreate(internalProfileId: Int) {
     this.profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()
-    getProfileData().observe(activity, Observer<StoryTextSize> { result ->
+    retrieveStoryTextSize().observe(fragment, Observer<StoryTextSize> { result ->
       (activity as DefaultFontSizeStateListener).onDefaultFontSizeLoaded(result)
```

Comment:
I was actually expecting that the manager fragment would just go ahead and resize the text itself, rather than blocking loading the exploration fragment. Shouldn't we eventually be able to resize text size from within the player? Resizing the font scale here simplifies the solution a bit, and should automatically work with that later feature once we finish hooking up the overflow menu part of it.

Reply:
@BenHenning Adding conversation context
> Here's what I was thinking: https://gist.github.com/BenHenning/0eb3e6139bcf2bd4bb790e323830b92f. The idea 
> is to have a fragment whose job is to just observe a data provider & then enable the activity to add/remove 
> fragments` or otherwise change its UI based on those results. I'm not sure if I explained this well--does it make sense to you?

@BenHenning We need to do resize within onAttach, Hence manager fragment just observe a data provider & then enable the activity to add fragment and helps to update in the onAttach of ExplorationFragment.

Reply:
@BenHenning  If I'm scaling after retrieveStoryTextSize then as initially informed scaling is not working as expected, for eg: on configuration change scale changes, if user change scale more than three times it won't work and so on.